Remotion/Web/Development.Analyzers.Tests/RMWEB0001_WebStringAnalyzerTests.cs

```diff
@@ -0,0 +1,254 @@
+﻿// This file is part of the re-motion Core Framework (www.re-motion.org)
+// Copyright (c) rubicon IT GmbH, www.rubicon.eu
+//
+// The re-motion Core Framework is free software; you can redistribute it
+// and/or modify it under the terms of the GNU Lesser General Public License
+// as published by the Free Software Foundation; either version 2.1 of the
+// License, or (at your option) any later version.
+//
+// re-motion is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with re-motion; if not, see http://www.gnu.org/licenses.
+//
+using System;
+using System.Text;
+using System.Web.UI;
+using Gu.Roslyn.Asserts;
+using NUnit.Framework;
+using Remotion.Web;
+using Web.Development.Analyzers;
+
+[assembly: MetadataReference (typeof (object))]
+[assembly: MetadataReference (typeof (WebString))]
+[assembly: MetadataReference (typeof (StringBuilder))]
+[assembly: MetadataReference (typeof (HtmlTextWriter))]
+
+namespace Remotion.Web.Development.Analyzers.Tests
+{
+  [TestFixture]
+  public class RMWEB0001_WebStringAnalyzerTests
+  {
+    [Test]
+    public void StringBuilder_WithWebString_Diagnostic ()
+    {
+      var analyzer = new RMWEB0001_WebStringAnalyzer();
+      const string input = @"
+using System.Text;
+using Remotion.Web;
+public class A
+{
+  public void Test()
+  {
+    var sb = new StringBuilder();
+    ↓sb.Append(WebString.CreateFromText(""test""));
```

Comment:
Would the tests be more precise if we moved the construction of WebString outside of the method call and just used a variable? After all, we don't want to detect if someone uses WebString.CreateFromText("...") but rather if someone uses WebString.

Reply:
Knowing how the analyzer is implemented, no. but it might be a good idea to ensure future refactorings produce the expected result.